### PR TITLE
Adds stopPort to getPort Api

### PR DESCRIPTION
**Pull request by Prithvirajbilla**
- Adds stopPort option to getPort Api
  - This enables getPort api to find open port in the range basePort - stopPort/
- also includes some code boyscouting

### CODE REVIEW

File: test/port-finder-test.js [hunk 1]
```diff
@@ -46,6 +46,35 @@ vows.describe('portfinder').addBatch({
...
+          // stopPort: 32774 is greater than available port 32773 (32768 + 5)
+          setTimeout(function() {
+            portfinder.getPort({ stopPort: 32774 }, this.callback);
+          }.bind(this), 3000); //wait for cleanup of bound hosts.
+        },
+        "should respond with the first free port (32773) less than provided stopPort": function(err, port) {
+          if (err) { debugVows(err); }
+          assert.isTrue(!err);
+          assert.equal(port, 32773);
+        }
```
Comment (eriktrom) [inline test/port-finder-test.js]: i think the socket is not fully closed by the time these are running for port 32773 - that's what those setTimeouts of 3 seconds are for -- its hacky i know - if you know vows and have a better way, by all means... 

it passes locally i assume?

Comment (eriktrom) [reply test/port-finder-test.js]: also check https://github.com/indexzero/node-portfinder/blob/master/test/helper.js - we start some httpServer's in there, might be messing with your stopPort

I can't recall, but will look deeper before we merge for sure

File: lib/portfinder.js [hunk 2]
```diff
@@ -88,6 +92,15 @@ exports.getPort = function (options, callback) {
...
     callback = options;
     options = {};
+
+  }
+
+  options.port   = Number(options.port) || Number(exports.basePort);
+  options.host   = options.host    || null;
+  options.stopPort = Number(options.stopPort) || Number(exports.highestPort)
+
+  if(!options.startPort) {
```
[the remark this answers is not in the document]
Comment (Prithvirajbilla) [reply lib/portfinder.js]: a) Done. 
b) Done. But question. If a user provides a port less than 1024, then we will not able to bound to them and eventually  program will return a port greater than 1024. So is this really needed?
c) Add one more check to guard against startPort is less than stopPort. 

Thanks for quick review.

File: test/port-finder-test.js (hunk 1 above)
Comment (Prithvirajbilla) [reply test/port-finder-test.js]: Yes, they are all passing locally.

Comment (Prithvirajbilla) [reply test/port-finder-test.js]: Yeah. you are right. All the tests were run asynchronously when test servers are spun up and there is a race condition in the tests. I think the way to fix this is to have all the tests spun their own test services which are not colliding. I'd like to do this as a followup.

Comment (eriktrom) [reply test/port-finder-test.js]: >  I think the way to fix this is to have all the tests spun their own test services which are not colliding. I'd like to do this as a followup.

agreed, and sounds awesome, thanks

File: lib/portfinder.js [hunk 3]
```diff
@@ -88,6 +92,22 @@ exports.getPort = function (options, callback) {
...
+
+  }
+
+  options.port   = Number(options.port) || Number(exports.basePort);
+  options.host   = options.host    || null;
+  options.stopPort = Number(options.stopPort) || Number(exports.highestPort);
+
+  if(!options.startPort) {
+    options.startPort = Number(options.port);
+    if(options.startPort < 1024) {
```
Comment (eriktrom) [inline lib/portfinder.js]: https://github.com/indexzero/node-portfinder/issues/65 filed due to the above line. @Prithvirajbilla do u mind making a quick pr - or i can but starting work in like.. now so would be in a few hours

Comment (eriktrom) [reply lib/portfinder.js]: https://github.com/indexzero/node-portfinder/pull/66

Comment (Prithvirajbilla) [reply lib/portfinder.js]: I saw your PR. Thanks for getting to this quickly.  I will clean this up and will add a test for this special case (port 80) when I get time.

Comment (eriktrom) [reply lib/portfinder.js]: LOL - yeah you'd be surprised how many upstream libs use this thing :) no worries, happens.

**Pull request merged**